cpp/dolfin/io/XDMFFile.cpp

```diff
@@ -40,10 +40,47 @@ using namespace dolfin;
 using namespace dolfin::io;
 
 //-----------------------------------------------------------------------------
-XDMFFile::XDMFFile(MPI_Comm comm, const std::string filename)
-    : _mpi_comm(comm), _filename(filename), _counter(0),
-      _xml_doc(new pugi::xml_document)
+XDMFFile::XDMFFile(MPI_Comm comm, const std::string filename,
+                   const std::string file_mode)
+    : _mpi_comm(comm), _h5_id(-1), _filename(filename),
+      _xml_doc(new pugi::xml_document), _file_mode(file_mode)
 {
+
+  if (!(_file_mode == "a" or _file_mode == "r" or _file_mode == "w"
+        or _file_mode == "wb" or _file_mode == "ab"))
+  {
+    throw std::runtime_error("Unknown file mode used in XDMFFile.");
+  }
+
+  // FIXME: Encoding serves just as proxy here, not really needed to keep it
+  // For binary modes set HDF5 encoding
+  if (_file_mode == "wb" or _file_mode == "ab")
+  {
```

Comment:
Remove braces - we tend to use one line for short control statements and bodies.